### ISSUE TITLE
Get-UserProfiles freezes script if saved into variable and domain account has been removed

### ISSUE THREAD

**Issue by CandymanRabbit** [PSAppDeployToolkit/PSAppDeployToolkit#798]
**Describe the bug**
When using Get-UserProfiles and saving output into a variable, script tends to freeze if there is deleted domain account.

**To Reproduce**
1. Create new domain account
2. Login with it to get profile created.
3. Delete domain account
4. Use 
```
[string[]]$UserProfiles = Get-UserProfiles | Select-Object 'ProfilePaths' 
foreach ($UserProfile in $UserProfiles) { Write-Log -Message "Yay, $UserProfile found" }
```

The problem is, when ConvertTo-NTAccountOrSID function is called and it catches SID translation error, it will try to resolve the error as well causing this function to freeze the script if Get-UserProfiles function output is saved to a variable.

Workaround is to remove ```r`n$(Resolve-Error)``` from ConvertTo-NTAccountOrSID
```Write-Log -Message "Failed to convert $msg. It may not be a valid account anymore or there is some other problem. `r`n$(Resolve-Error)" -Severity 3 -Source ${CmdletName}```

**Comment by That-Annoying-Guy** [comment]
Confirmed in Discourse forums https://discourse.psappdeploytoolkit.com/t/bug-in-get-userprofiles/5519

**Comment by DanGough** [comment]
Related issue: 

https://github.com/PSAppDeployToolkit/PSAppDeployToolkit/issues/865

**Comment by mjr4077au** [comment]
This should be addressed by way of the fix outlined in #865.
